### ISSUE TITLE
variable mount is null on every page

### ISSUE THREAD

**Issue by lostika86** [statamic/cms#3625]
## Bug Description
Mount variable is missing from collections entries. It is attached as in the docs. Last time worked correctly,
## How to Reproduce
1. Create new collection, and define the mount

```title: 'No More Mount'
sites:
  - default
  - hungarian
template: default
layout: layout
mount: 28124839-6062-43bd-8e4e-ca7a11af3b7a
revisions: false
route: '{{mount}}/{{slug}}'
sort_dir: asc
date_behavior:
  past: public
  future: private 
```

2. Create an entry and go to url (`{{mount}}/{{slug}}`)

## Extra Detail

Few weeks ago worked correctly
## Environment

Statamic 3.1.11 Pro
Laravel 8.40.0
PHP 7.3.12
doublethreedigital/duplicator 1.3.1

**Install method (choose one):**
- Fresh install from `statamic/statamic`

**Comment by robdekort** [comment]
Hmmm, seems to work fine here with this syntax: `/{mount}/{slug}`.

**Comment by lostika86** [comment]
Can you reach the mount variable? {{ mount | dump }} is null for me. Debugbar shows on mount null too

**Comment by robdekort** [comment]
The mount is set on a collection level and I don't think you can reach it in your template. But your issue is about the actual route giving a 404 right? 

So when you visit `yoursite.test/mount-slug/entry-slug` you got a 404 is what I got from your initial issue. 

I noticed you used a different syntax. Have you tried my suggestion?

**Comment by jasonvarga** [comment]
`mount` will be the handle of the collection on the entry that has been mounted to it.

Seems to work fine.

In your example, within entry `28124839-6062-43bd-8e4e-ca7a11af3b7a`, `{{ mount }}` will be the handle of whatever collection's yaml file you pasted.

**Comment by robdekort** [comment]
I'm not sure what the real issue is tbh. Can you explain in detail what you are trying to achieve and what you think is the issue?

**Comment by lostika86** [comment]
> The mount is set on a collection level and I don't think you can reach it in your template. But your issue is about the actual route giving a 404 right?
> 
> So when you visit `yoursite.test/mount-slug/entry-slug` you got a 404 is what I got from your initial issue.
> 
> I noticed you used a different syntax. Have you tried my suggestion?

The route works fine. I want to reach it in my template, there is a variable, `mount:null` if I check variables on debugbar

**Comment by lostika86** [comment]
I want to simply use in my template for example: `{{ mount:title }}`

**Comment by lostika86** [comment]
another way to explain: 
current entry belongs into `categories` collection
![image](https://user-images.githubusercontent.com/10046659/117714163-d59ebd80-b1d6-11eb-82e0-de8ca7c70d88.png)
the `mount` variable is `null`
![image](https://user-images.githubusercontent.com/10046659/117714348-1565a500-b1d7-11eb-872d-22cca132eccf.png)

**Comment by lostika86** [comment]
> I'm not sure what the real issue is tbh. Can you explain in detail what you are trying to achieve and what you think is the issue?

1. Create a new collection, dont forget to setup mount. 
2. Create a new entry 'Mountains' in collection
3. Go to template, where you can reach/use 'Mountains' entry
4. You will see in debugbar available variables for entry, but variable mount is null

**Comment by robdekort** [comment]
I think it's supposed to work like that. It's only available on the entry where you mounted the collection on.
I've been in multiple use cases where I need to get a collection's mount entry URL so you can do stuff like "Go back to all news" / "Go back to all mountains". 

There's no native tag for this. There is a FR out: https://github.com/statamic/ideas/issues/301 and a solution in the comments (link to forum).

Hope this helps!

**Comment by robdekort** [comment]
Closing this, feel free to reopen if I'm wrong.

**Comment by edalzell** [comment]
I'm in a multi-site and the `mount` var is null on `career` entry pages (which are mounted to a `careers` page).

**Comment by jasonvarga** [comment]
This has been reverted in #9145 as it causes some issues. We'll revisit it in another version.

**Comment by isaackearl** [comment]
I'm running into this same issue.  If the mount is null, how can I access the page data the collection is mounted on?  I too want to be able to do mount:title or something similar.

**Comment by 925dk** [comment]
I'm looking for the same. Seems no good way to get say the mount title. Doing this as work-around:

```
{{ if {mount_url:{collection}} }}
    {{ get_content from="{{mount_url:{collection}} | replace("/" + site:handle + "/", "/")}" }}
        {{ title }}
    {{ /get_content }}
{{ /if }}
```

The string replacement stuff seems necessary also, since `mount_url` returns full path for multisite, and `get_content` needs it without.

**Comment by carstenjaksch** [comment]
The workaround from this [issue](https://github.com/statamic/ideas/issues/301#issuecomment-803908974) is not available anymore.

I want to do the same thing, any updates on this?

**Comment by duncanmcclean** [comment]
@carstenjaksch Have you seen @925dk's workaround above? Does it work for you?

**Comment by carstenjaksch** [comment]
@duncanmcclean Yeah, I tried and mentioned it in this discussion: https://github.com/statamic/cms/discussions/11237

It works for now, I was just curious if the official solution would return.

**Comment by duncanmcclean** [comment]
The previous fix broke a bunch of things so it had to be reverted. We might fix it another way in the future, but I don't have an ETA on that.

**Comment by FlxRobole** [comment]
Bump - We are facing the same issue with a similar twist: When configuring a "mount" entry on a collection, all items of that collection will indeed get the correct (and site-/locale-dependent!) url-prefix of the "mounted" entry, but the "parent" and "mount" value of these items will remain empty (in 90% of the cases... really strange). Also, the "Add/Edit" link that is displayed next to an entry that was linked as the "mount" value on another collection will only be visible in the site/locale it was linked in.

**Comment by j3ll3yfi5h** [comment]
Same or similar issue here: Randomly, the mount is missing on some collections. Feels like the entries with the `{mount}` in the urls are stached before the mounted entry is available. Could this be possible? In what order is the stache generated?

Also not nice to have them in the google index with the mount and one day without:
Correct and indexed: https://g-m.swiss/maerkte/sport-mehrzweckanlagen
Right now: 404 -> available under https://g-m.swiss/sport-mehrzweckanlagen
